src/crimson/os/seastore/transaction_manager.h

```diff
@@ -551,7 +551,7 @@ class TransactionManager : public SegmentCleaner::ExtentCallbackInterface {
 	*segment_cleaner,
 	*sm,
 	*lba_manager,
-	*journal,
+	journal->as_seg_journal(),
```

Comment:
It's not obvious to me why this needs to be a SegmentJournal.  SegmentedAllocator is using this instance to get the sequence id, not to perform the write.  Passing a CBJournal would be fine, and necessary in cases where are writing ool extents to a zns device in a configuration with a CBJournal.

Reply:
Rebased. Due to recent changes, the journal is no longer the input argument.